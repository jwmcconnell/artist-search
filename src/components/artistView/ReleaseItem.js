import React from 'react';
import PropTypes from 'prop-types';

const styles = {
  item: {
    textAlign: 'center',
    width: '300px',
    height: '300px',
    margin: '5px'
  },
  image: {
    width: '250px',
    height: '250px'
  }
};

const ReleaseItem = ({ release }) => {
  const image = release['cover-art-archive'].front 
    ? `http://coverartarchive.org/release/${release.id}/front` 
    : '/src/assets/placeholder.jpg';
  return (
    <li style={styles.item}>
      <h3>{release.title}</h3>
      <img src={image} style={styles.image}/>
    </li>
  );
};

ReleaseItem.propTypes = {
  release: PropTypes.object.isRequired
};

export default ReleaseItem;
