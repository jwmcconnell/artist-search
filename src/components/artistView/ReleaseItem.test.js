import React from 'react';
import { shallow } from 'enzyme';
import ReleaseItem from './ReleaseItem';

describe('ReleaseItem component', () => {
  it('renders ReleaseItem', () => {
    const wrapper = shallow(<ReleaseItem release={{ 'cover-art-archive': { font: null } }}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
