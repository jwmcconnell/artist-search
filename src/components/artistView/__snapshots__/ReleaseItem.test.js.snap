// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleaseItem component renders ReleaseItem 1`] = `
<Link
  to="/release/undefined/undefined"
>
  <li
    style={
      Object {
        "height": "300px",
        "margin": "5px",
        "textAlign": "center",
        "width": "300px",
      }
    }
  >
    <h3 />
    <img
      src="/src/assets/placeholder.jpg"
      style={
        Object {
          "height": "250px",
          "width": "250px",
        }
      }
    />
  </li>
</Link>
`;
